src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/aggregators/PersistableSlidingWindow.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.aggregators;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.persistence.FileRotate;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Iterator;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.LineIterator;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * PersistableSlidingWindow is a SlidingWindow which can have its data written to and read from disk
+ */
+public class PersistableSlidingWindow extends SlidingWindow<SlidingWindowData> {
+  private static final Logger LOG = LogManager.getLogger(PersistableSlidingWindow.class);
+  private static final ObjectMapper objectMapper = new ObjectMapper();
+
+  protected static final String SEPARATOR = "\n";
+
+  private Path pathToFile;
+
+  public PersistableSlidingWindow(int slidingWindowSize,
+                                  TimeUnit timeUnit,
+                                  Path filePath) {
+    super(slidingWindowSize, timeUnit);
+    this.pathToFile = filePath;
+    try {
```

Comment:
should we do this check within the constructor itself ? I think it would make more sense to perform this check during writing operation.

Reply:
I don't expose the load function to consumers. This class loads previous data exactly once upon construction. It doesn't need to load during write/next operations.

Reply:
The reason I create the file here is because I need to do the check before I load. The reason this logic shouldn't live inside write() is because it would waste every single instruction to check beyond the first

Reply:
I thought about it, and I see what you were getting at. I preserved the load in the constructor, but I removed the create file as it's unnecessary.